lib/aligned-lockable-buffer.ts

```diff
@@ -0,0 +1,66 @@
+import { getAlignedBuffer } from '@ronomon/direct-io';
+import RWMutex = require('rwmutex');
+
+export interface AlignedLockableBuffer extends Buffer {
+	alignment: number;
+	lock: () => Promise<() => void>;
+	rlock: () => Promise<() => void>;
+	slice: (start?: number, end?: number) => AlignedLockableBuffer;
+}
+
+function attachMutex(
+	buf: Buffer,
+	mutex: RWMutex,
+	alignment: number,
+): AlignedLockableBuffer {
+	const buffer = buf as AlignedLockableBuffer;
+	buffer.alignment = alignment;
+	buffer.lock = mutex.lock.bind(mutex);
+	buffer.rlock = mutex.rlock.bind(mutex);
+	const bufferSlice = buffer.slice.bind(buffer);
+	buffer.slice = (...args) => {
+		const slice = bufferSlice(...args);
+		return attachMutex(slice, mutex, alignment);
+	};
+	return buffer;
+}
```

Comment:
This could benefit from the same improvements I mentioned for a standard lockable buffer